src/openai/azure/_azuremodels.py

```diff
@@ -1,5 +1,58 @@
-from typing import TypedDict
+from typing import List, Optional
+from typing_extensions import TypedDict
+from openai.types.chat import ChatCompletion, ChatCompletionChunk, ChatCompletionMessage
+from openai.types.chat.chat_completion import Choice as ChatChoice
+from openai.types.chat.chat_completion_chunk import ChoiceDelta, Choice as ChatChoiceDelta
+from openai._models import BaseModel
+
 
 class ChatExtensionConfiguration(TypedDict):
     type: str
     parameters: object
+
+
+class ContentFilterResult(BaseModel):
+    severity: str
+    filtered: bool
+
+
+class ContentFilterResults(BaseModel):
+    hate: ContentFilterResult
+    self_harm: ContentFilterResult
+    violence: ContentFilterResult
+    sexual: ContentFilterResult
+
+class PromptFilterResult(BaseModel):
+    prompt_index: int
+    content_filter_results: ContentFilterResults
+
+class AzureChatExtensionsMessageContext(BaseModel):
+    messages: List[ChatCompletionMessage]
+
+
+class AzureChatCompletionMessage(ChatCompletionMessage):
+    context: Optional[AzureChatExtensionsMessageContext] = None
+
+
+class AzureChatCompletionChoice(ChatChoice):
+    content_filter_results: Optional[ContentFilterResults] = None
+    message: AzureChatCompletionMessage  # TODO typing hates this
```

Comment:
Yeah, subclassing invariant types can indeed be an issue here. This will be a gnarly issue I suspect.

Reply:
I've sprinkled in some type: ignores for now...